internal/plugin/transport.go

```diff
@@ -70,7 +84,31 @@ func NewTransportHandle(name string, t envelope.Transport) (Handle, error) {
 	if handshake.APIVersion != api.APIVersion {
 		return nil, errHandshakeFailed{
 			Name:   name,
-			Reason: errVersionMismatch{Want: api.APIVersion, Got: handshake.APIVersion},
+			Reason: errAPIVersionMismatch{Want: api.APIVersion, Got: handshake.APIVersion},
```

Comment:
Thanks to your PR description I understand that .Version is the version of the thriftrw library that was used during compilation of the plugin itself and .APIVersion is the plugin API version itself.

So  what is really the goal of checking both APIVersion and thriftrw Version? APIVersion should be sufficient isn't? As long as you speak the right API, the interpretation of it on both sides should stay the same no? And if something changes in the thriftrw API enough, it warrants a new plugin API, ensuring only updated plugins will work with the new library.

On the other hand since the plugin API will never change without changing the thriftrw version, maybe .APIVersion is not needed after all, as we can assume the thriftrw API includes the plugin API.

A the very least, maybe rename handshake.Version to handshake.ThriftRWVersion or .LibVersion, just to make it a tad easier to read.